pkg/capacityscheduling/capacity_scheduling.go

```diff
@@ -0,0 +1,841 @@
+
+// ElasticQuotaSnapshot stores the snapshot of elasticQuotas.
+type ElasticQuotaSnapshotState struct {
+	elasticQuotaInfos ElasticQuotaInfos
+}
+
+// Clone the ElasticQuotaSnapshot state.
+func (s *ElasticQuotaSnapshotState) Clone() framework.StateData {
+	return &ElasticQuotaSnapshotState{
+		elasticQuotaInfos: s.elasticQuotaInfos.clone(),
+	}
+}
+
+var _ framework.PreFilterPlugin = &CapacityScheduling{}
+var _ framework.PostFilterPlugin = &CapacityScheduling{}
+var _ framework.ReservePlugin = &CapacityScheduling{}
+
+const (
+	// Name is the name of the plugin used in Registry and configurations.
+	Name = "CapacityScheduling"
+
+	// preFilterStateKey is the key in CycleState to NodeResourcesFit pre-computed data.
+	preFilterStateKey       = "PreFilter" + Name
+	ElasticQuotaSnapshotKey = "ElasticQuotaSnapshot"
+)
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (c *CapacityScheduling) Name() string {
+	return Name
+}
+
+// New initializes a new plugin and returns it.
+func New(obj runtime.Object, handle framework.FrameworkHandle) (framework.Plugin, error) {
+	args, ok := obj.(*config.CapacitySchedulingArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type CapacitySchedulingArgs, got %T", obj)
+	}
+	kubeConfigPath := args.KubeConfigPath
+
+	c := &CapacityScheduling{
+		frameworkHandle:   handle,
+		elasticQuotaInfos: NewElasticQuotaInfos(),
+		pdbLister:         getPDBLister(handle.SharedInformerFactory()),
+	}
+
+	restConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	client, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	schedSharedInformerFactory := schedinformer.NewSharedInformerFactory(client, 0)
+	c.elasticQuotaLister = schedSharedInformerFactory.Scheduling().V1alpha1().ElasticQuotas().Lister()
+	elasticQuotaInformer := schedSharedInformerFactory.Scheduling().V1alpha1().ElasticQuotas().Informer()
+	elasticQuotaInformer.AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				switch t := obj.(type) {
+				case *v1alpha1.ElasticQuota:
+					return true
+				case cache.DeletedFinalStateUnknown:
+					if _, ok := t.Obj.(*v1alpha1.ElasticQuota); ok {
+						return true
+					}
+					utilruntime.HandleError(fmt.Errorf("cannot convert to *v1.Pod: %v", obj))
+					return false
+				default:
+					utilruntime.HandleError(fmt.Errorf("unable to handle object in %T", obj))
+					return false
+				}
+			},
+			Handler: cache.ResourceEventHandlerFuncs{
+				AddFunc:    c.addElasticQuota,
+				UpdateFunc: c.updateElasticQuota,
+				DeleteFunc: c.deleteElasticQuota,
+			},
+		})
+
+	schedSharedInformerFactory.Start(nil)
+	if !cache.WaitForCacheSync(nil, elasticQuotaInformer.HasSynced) {
+		return nil, fmt.Errorf("timed out waiting for caches to sync %v", Name)
+	}
+
+	podInformer := handle.SharedInformerFactory().Core().V1().Pods().Informer()
+	podInformer.AddEventHandler(
```

Comment:
Yes. We will add the corresponding controller logic for the `status` field of ElasticQuota. This will be used by other operator or users. But in order to prevent data synchronization delay, I think it's better to maintain the status in the scheduler itself also.